io.c

```diff
@@ -1975,7 +1975,7 @@ io_fwrite(VALUE str, rb_io_t *fptr, int nosync)
     if (converted)
         OBJ_FREEZE(str);
 
-    tmp = rb_str_tmp_frozen_acquire(str);
```

Comment:
Would it make sense just to change `rb_str_tmp_frozen_acquire` rather than introducing a new method?